lib/internal/event_target.js

```diff
@@ -233,7 +239,24 @@ class EventTarget {
     initEventTarget(this);
   }
 
-  [kNewListener](size, type, listener, once, capture, passive) {}
+  [kNewListener](size, type, listener, once, capture, passive) {
+    if (this[kMaxEventTargetListeners] > 0 &&
+        size > this[kMaxEventTargetListeners] &&
+        !this[kMaxEventTargetListenersWarned]) {
+      this[kMaxEventTargetListenersWarned] = true;
+      // No error code for this since it is a Warning
+      // eslint-disable-next-line no-restricted-syntax
+      const w = new Error('Possible EventTarget memory leak detected. ' +
+                          `${size} ${type} listeners ` +
+                          `added to ${inspect(this, { depth: -1 })}. Use ` +
+                          'events.setMaxListeners() to increase limit');
```

Comment:
It's `events.setMaxListeners()` here but `EventEmitter.setMaxListeners()` in the docs. Should it be consistent or is there a good reason for the discrepancy?

Reply:
Or maybe it's the instance method `event.setMaxListeners()`? Maybe either one?

Reply:
That's really due to the oddity that is `require('events')` ...

We often see:
```js
const events = require('events')
```

Or
```js
const EventEmitter = require('events')
```

Or

```
const { EventEmitter } = require('events')
```

etc

All of which give you the same thing.

And there's inconsistency in the events.md doc as is as you've pointed out. I'll open a separate PR after this lands to reorganize events.md a bit and get more consistency.